cmd/kubeadm/app/cmd/init.go

```diff
@@ -307,6 +307,13 @@ func newInitData(cmd *cobra.Command, options *initOptions) (initData, error) {
 
 // runInit executes master node provisioning
 func runInit(i *initData, out io.Writer) error {
+	// First off, configure the kubelet. In this short timeframe, kubeadm is trying to stop/restart the kubelet
+	// Try to stop the kubelet service so no race conditions occur when configuring it
+	if !i.dryRun {
```

Comment:
I don't see why you would stop the kubelet before the preflight checks.  If the preflight checks fail you've now put the kubelet into a non-functioning state.

Reply:
The preflight checks [assumes the kubelet is not yet listening on 10250](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L858). 

This is because when we install the kubeadm package we also install a non-functional kubelet systemd file and restart the kubelet to put it into a broken state. This PR will install a *functioning* kubelet and will therefore fail the preflight check's assumption.

Reply:
that was the wrong link, the correct link is: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L997